### PR TITLE
Use separate AMI artifacts for deploying and as the base_ami for mess…

### PR DESCRIPTION
**Pull request by cpennington**
…aging

### CODE REVIEW

File: edxpipelines/patterns/edxapp.py [hunk 1]
```diff
@@ -248,22 +248,29 @@ def generate_deploy_stages(pipeline_name_build, auto_deploy_ami=False):
...
             config['asgard_api_endpoints'],
             config['asgard_token'],
             config['aws_access_key_id'],
             config['aws_secret_access_key'],
-            ami_file_location,
+            built_ami_file_location,
             manual_approval=not auto_deploy_ami
         )
 
+        base_ami_file_location = utils.ArtifactLocation(
```
Comment (macdiesel) [inline edxpipelines/patterns/edxapp.py]: Where does this get used?

Comment (doctoryes) [inline edxpipelines/patterns/edxapp.py]: This definition isn't used anywhere - why define it?

File: edxpipelines/patterns/edxapp.py [hunk 2]
```diff
@@ -248,22 +248,29 @@ def generate_deploy_stages(pipeline_name_build, auto_deploy_ami=False):
     # Create the stage to deploy the AMI.
     #
     def builder(pipeline, config):
```
Comment (macdiesel) [inline edxpipelines/patterns/edxapp.py]: This looks correct to me.

File: edxpipelines/patterns/edxapp.py (hunk 1 above)
Comment (cpennington) [inline edxpipelines/patterns/edxapp.py]: Fixed. It's used below for messaging.

**Pull request merged**